modulemd/v2/modulemd-defaults.c

```diff
@@ -0,0 +1,309 @@
+/*
+ * This file is part of libmodulemd
+ * Copyright (C) 2017-2018 Stephen Gallagher
+ *
+ * Fedora-License-Identifier: MIT
+ * SPDX-2.0-License-Identifier: MIT
+ * SPDX-3.0-License-Identifier: MIT
+ *
+ * This program is free software.
+ * For more information on the license, see COPYING.
+ * For more information on free software, see <https://www.gnu.org/philosophy/free-sw.en.html>.
+ */
+
+
+#include <inttypes.h>
+#include "modulemd-defaults.h"
+#include "modulemd-defaults-v1.h"
+#include "private/modulemd-util.h"
+
+#define DEF_DEFAULT_NAME_STRING "__NAME_UNSET__"
+
+typedef struct
+{
+  gchar *module_name;
+} ModulemdDefaultsPrivate;
+
+G_DEFINE_ABSTRACT_TYPE_WITH_PRIVATE (ModulemdDefaults,
+                                     modulemd_defaults,
+                                     G_TYPE_OBJECT)
+
+enum
+{
+  PROP_0,
+
+  PROP_MDVERSION,
+  PROP_MODULE_NAME,
+
+  N_PROPS
+};
+
+static GParamSpec *properties[N_PROPS];
+
+
+ModulemdDefaults *
+modulemd_defaults_new (guint64 mdversion, const gchar *module_name)
+{
+  g_return_val_if_fail (mdversion && mdversion <= MD_DEFAULTS_VERSION_LATEST,
```

Comment:
Shouldn't this be `MD_DEFAULTS_VERSION_ONE`, since this would seem to imply that when a Defaultsv2 is added, it is no longer possible to create a DefaultsV1.
While I guess your intention would be to modify this later, maybe it's an idea to make this intention clear by either using `VERSION_ONE`, or adding a `MD_DEFAULTS_VERSION_LOWEST_SUPPORTED`?